src/components/DnsSettings.tsx

```diff
@@ -37,80 +49,124 @@ function getUpdatedSettings(settings: DnsSettings, action: Action): DnsSettings
   };
 }
 
-const DnsSettingsForm: FC<Props> = ({ settings, isEditor, onUpdate }) => {
-  const values = {
-    ...defaultValues,
-    ...settings.dns,
-  };
+const DnsSettingsForm: FC<Props> = ({ settings, isEditor, labels, onUpdate }) => {
+  const values = useMemo(
+    () => ({
+      ...defaultValues,
+      ...settings.dns,
+    }),
+    [settings.dns]
+  );
 
   const [showValidation, setShowValidation] = useState(false);
   const [showAdvanced, setShowAdvanced] = useState(false);
+  const [showDNSSettings, setShowDNSSettings] = useState(false);
+  const onValidateAnswerChange = useCallback(
+    (validations: DNSRRValidator | undefined) => {
+      const dns = getUpdatedSettings(values, { name: 'validateAnswerRRS', value: validations });
+      onUpdate({ dns }, labels);
+    },
+    // eslint-disable-next-line
+    []
```

Comment:
going out of your way here to not specify `onUpdate` as dependency :)

Reply:
Also `values`.

Whenever one has to disable eslint, something fishy might be going on. Because unless `onUpdate` and `values` change frequently, this function wouldn't get recreated anyway. And if they to change, wouldn't you want these handlers to get updated?